### DISCUSSION TITLE
Add binary sensor device_class: fan

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/architecture#921] [Entity Models] [answered]
seems too obvious to miss hence my issues in creating a binary_sensor which I set using it without even checking at first...
using replacement `device_class: heat` for now, though that is really not the same, and has other real life uses.

checking on https://developers.home-assistant.io/docs/core/entity/binary-sensor#available-device-classes however revealed the system was right, 

```
Logger: homeassistant.config
Source: config.py:982 
First occurred: 08:45:56 (1 occurrences) 
Last logged: 08:45:56

Invalid config for [template]: expected BinarySensorDeviceClass or one of 'battery', 'battery_charging', 'carbon_monoxide', 'cold', 'connectivity', 'door', 'garage_door', 'gas', 'heat', 'light', 'lock', 'moisture', 'motion', 'moving', 'occupancy', 'opening', 'plug', 'power', 'presence', 'problem', 'running', 'safety', 'smoke', 'sound', 'tamper', 'update', 'vibration', 'window' for dictionary value @ data['binary_sensor'][4]['device_class']. Got 'fan'. (See /config/packages/package_home_climate.yaml, line 247).
```

there is no such class...

my suggestion would be very simple, add the `device_class: fan`, and have it use the regular icons for `mdi:fan` and `mdi:fan-off`.

next also have that option listed in the frontend as available for show_as.

example config to illustrate the purpose:
```
      - unique_id: ventileer_patchboard
        state: >
          {% set udm = states('sensor.udm_pro_cpu_temperature')|float(0) %}
          {% set cpu = states('sensor.cpu_temp')|float(0) %}
          {% set buiten = states('sensor.buienradar_temperature')|float(0) %}
          {{udm > 45 and buiten > 20 or
            cpu > 50 or 
            udm > 55}}
        delay_on:
          minutes: 2
        delay_off:
          minutes: 2
        device_class: fan #heat
```

a bit more background:
this binary would be used to trigger an automation switching a regular smart switch connected to a fan. Also considered were a template switch, and even a template fan. None of those however facilitate this as easily as the binary_sensor/automation combo.

**Comment by MartinHjelmare** [comment]
Which integrations would benefit from this new binary sensor device class? Please list them. We normally require about a handful of integrations before adding a device class.

**Reply by Mariusthvdb** [reply]
based on what I would think it could be used for those would be at least the main binary_sensor, template, switch-as-x, template switch.
Ofc the show_as, but that would be Frontend using this new class.

**Reply by MartinHjelmare** [reply]
Are there any device integrating integrations that would benefit?

**Reply by Mariusthvdb** [reply]
I would think any integration that provides a binary regarding on/off state for fans. think zwave, zigbee, esphome, shelly and the likes. Overkiz and other for brand integrations in core. Even group.

**Reply by MartinHjelmare** [reply]
Normally a fan device is also controllable and will be exposed as a fan entity and not a binary sensor entity.

When I asked for integrations I meant integrations that today has a binary sensor that represents a fan, where the fan can't be exposed as a fan entity. Please search for actual examples and don't count hypothetical answers.

**Reply by Mariusthvdb** [reply]
My pardon, I didn't fully understand that.
I haven't found other fan providing integrations yet. 

My point is just that. With this template binary integration we can mimic a fan without it actually being a real device.

Just like we can set a binary to mimic a window or cover or any of the other available classes without it being a physical device itself.

**Reply by MartinHjelmare** [reply]
Until there are more device integrating integrations with this need, I'm not in favor of this proposal. It's not possible to judge the popularity of virtual devices.

**Reply by frenck** [reply, marked as answer]
I agree with Martin on this proposal.

Therefore going to friendly decline this proposal for now.

../Frenck
